acid/features/history/model.py

```diff
@@ -76,9 +76,13 @@ def get_for_pipeline(cls, pipeline):
             len(select(b for b in bs.builds)) > 0).sort_by(desc(cls.id))
 
     @classmethod
-    def get_filtered(cls, pipeline, branch='', build=''):
-        if branch not in cls.get_branches():
-            branch = ''
+    def get_filtered(cls, pipeline, branches, build=''):
+        for index, branch in enumerate(branches):
+            if branch not in cls.get_branches():
+                branches.pop(index)
+
+        if (len(branches) == 0):
+            branches = [branch for branch in cls.get_branches()]
```

Comment:
This line copy branches from `cls.get branches()` to new list by iterating over every element. Iterating is not required at all. Just assign result to variable.

Few lines above I made assignment: `all_branches = cls.get_branches()`
I can reuse it here `branches = all_branches`.